pkg/dynatrace/dynatrace.go

```diff
@@ -247,48 +272,51 @@ func (dynatraceApiClient *DynatraceApiClient) CreateDynatraceHttpMonitor(monitor
 	return monitorId, nil
 }
 
-func (dynatraceApiClient *DynatraceApiClient) ExistsHttpMonitorInDynatrace(monitorId string) (bool, error) {
-	path := ("/synthetic/monitors/")
-	resp, err := dynatraceApiClient.MakeRequest(http.MethodGet, path, "")
+func (dynatraceApiClient *DynatraceApiClient) ExistsHttpMonitorInDynatrace(clusterId string) (bool, error) {
+	existsHttpMonitorResponse, err := dynatraceApiClient.GetListDynatraceHttpMonitors(clusterId)
 	if err != nil {
 		return false, err
 	}
-	defer resp.Body.Close()
 
-	// Check if the response status code is OK
-	if resp.StatusCode != http.StatusOK {
-		return false, fmt.Errorf("failed to fetch monitor in Dynatrace. Status code: %d", resp.StatusCode)
-	}
+	countMonitors := len(existsHttpMonitorResponse.Monitors)
 
-	var existsHttpMonitorResponse ExistsHttpMonitorInDynatraceResponse
-	if err := json.NewDecoder(resp.Body).Decode(&existsHttpMonitorResponse); err != nil {
-		return false, fmt.Errorf("error parsing JSON: %w", err)
-	}
+	switch {
+	case countMonitors == 1:
+		return true, nil
 
-	for _, monitor := range existsHttpMonitorResponse.Monitors {
-		if monitor.EntityId == monitorId {
-			return true, nil
+	case countMonitors == 0:
+		return false, nil
+
+	case countMonitors > 1:
+		for i := 1; i < countMonitors; i++ {
+			if err := dynatraceApiClient.DeleteDynatraceMonitorByCluserId(clusterId); err != nil {
+				return false, fmt.Errorf("failed to delete monitors for cluster id %s: %w", clusterId, err)
+			}
 		}
+		return true, nil
 	}
+
 	return false, nil
 }
 
-func (dynatraceApiClient *DynatraceApiClient) DeleteDynatraceHttpMonitor(monitorId string) error {
-	path := fmt.Sprintf("/synthetic/monitors/%s", monitorId)
-
-	resp, err := dynatraceApiClient.MakeRequest(http.MethodDelete, path, "")
+func (dynatraceApiClient *DynatraceApiClient) DeleteDynatraceMonitorByCluserId(clusterId string) error {
+	existsHttpMonitorResponse, err := dynatraceApiClient.GetListDynatraceHttpMonitors(clusterId)
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
 
-	//monitor already deleted
-	if resp.StatusCode == http.StatusNotFound {
-		return nil
-	}
+	for _, monitor := range existsHttpMonitorResponse.Monitors {
```

Comment:
It doesn't look like a logger exists here, but this would be a good spot to `if len(monitors) == 0 { log.Info("No monitors found for cluster") }`

Reply:
Do you think I should wire up `logr.Logger` for this purpose? I have a couple of places that would need that passed though.

Reply:
Not as part of this PR no, but would be good in the future